GitUI/WindowsJumpListManager.cs

```diff
@@ -146,6 +158,13 @@ public void CreateJumpList(IntPtr windowHandle, WindowsThumbnailToolbarButtons b
                 CreateTaskbarButtons(windowHandle, buttons);
             }, nameof(CreateJumpList));
 
+            if (ToolbarButtonsCreated && _deferredAddToRecent != null)
+            {
+                var toAdd = _deferredAddToRecent;
+                _deferredAddToRecent = null;
+                AddToRecent(toAdd);
```

Comment:
I think we do not need thread-safety here (if, `Interlocked.Exchange` should be used).
So the temporary variable `toAdd` can be omitted.

Reply:
Yes, this is an "in between" approach. Not quite thread-safe, as it's no big deal if by extraordinary `AddToRecent` is called twice, but still much less likely if the time-window doesn't include an IO-bound call. Not convinced it's worth the hassle, though...